torch/distributed/distributed_c10d.py

```diff
@@ -243,13 +243,27 @@ def __init__(self, backend: Union[str, Backend]):
                 "cuda": backend_val,
             }
         else:
-            # custom backend string in format of "{device_type1}:{backend1},{device_type2}:{backend2}"
-            # TODO
-            pass
-
-        required_devices = ["cpu", "cuda"]
-        for device in required_devices:
-            assert device in self.device_backend_map
+            # make sure the backend string is in the correct format:
+            # "{device_type1}:{backend1},{device_type2}:{backend2}"
+            pattern = r'^[a-zA-Z0-9]+:[a-zA-Z0-9]+(,[a-zA-Z0-9]+:[a-zA-Z0-9]+)*$'
```

Comment:
Maybe more comments for this syntax if you can?

An idea for avoiding using `re` at all: 
just go ahead to `split` the string by `,` and then split by `:` (as your code below does). If the returned split list's length `!= 2`, that means either `:` is not found or there are too many `:`.

Reply:
Yeah I agree the check can be done that way. I think regex might be a little more manageable and result in less if conditions. It also limits the inputs to alphanumeric characters. For example if user does something accidentally like `cpu;:gloo` then it would error with regex (due to the `;`) but we would need another case to check for special characters. If you want to checks for each `:` pair I can add that instead, lmk!

Updated the comments as well.

Reply:
> It also limits the inputs to alphanumeric characters

Then something like "my_awesome_backend" would not work?

Reply:
hmm thats right, do you think it is necessary?